app.go

```diff
@@ -73,32 +70,28 @@ func (a *App) Serve() error {
 	// if configured to do so, start the workers
 	if !a.WorkerOff {
 		go func() {
-			err := a.Worker.Start(ctx)
-			if err != nil {
+			if err := a.Worker.Start(ctx); err != nil {
 				a.Stop(err)
 			}
 		}()
 	}
 
-	var err error
-
 	if strings.HasPrefix(a.Options.Addr, "unix:") {
 		// Use an UNIX socket
 		listener, err := net.Listen("unix", a.Options.Addr[5:])
 		if err != nil {
 			return a.Stop(err)
 		}
 		// start the web server
-		err = server.Serve(listener)
-	} else {
-		// Use a TCP socket
-		server.Addr = a.Options.Addr
-
-		// start the web server
-		err = server.ListenAndServe()
+		if err = server.Serve(listener); err != nil {
+			return a.Stop(err)
+		}
```

Comment:
There is a missing `return nil` here, isn't it?

Reply:
Maybe. ;)